### ISSUE TITLE
`--count` silently ignores `--out` on `package`

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3758]
`package <id> --count` writes to stdout and never creates the `--out` file, while the ordinary render honors it. The file is not created and no diagnostic is emitted, so a script redirecting output gets an empty/absent file at exit 0.

Pre-existing and independent of the count shape. Measured on `origin/main` at `5446864e6`:

```
package Markout -S "Manifest,Signature" --count --out /tmp/p1.txt   # map to stdout, /tmp/p1.txt absent
package Markout -S Manifest --count --out /tmp/p2.txt               # scalar 4 to stdout, /tmp/p2.txt absent
package Markout -S "Manifest" --out /tmp/p4.txt                     # ordinary render: file written, stdout empty
```

Head of #3754 behaves identically to base for both, so this is not caused by that PR; it was found while reviewing it (the new bare-`-S` count map inherits the same behavior).

Either honor `--out` for `--count`, or reject the combination rather than silently dropping the option.
